apis/metacontroller/v1alpha1/types_generic.go

```diff
@@ -167,7 +167,13 @@ type GenericControllerResource struct {
 	// Include the resource if resource selector matches
 	//
 	// This is ANDed with other selectors if present
-	ResourceSelector *ResourceSelector `json:"resourceSelector,omitempty"`
+	//
+	// NOTE:
+	//	This is an advanced selector & can be used to perform
+	// matches on complex selection criterias against combinations
+	// of labels, annotations, name, namespace, target object,
+	// path & slice values.
+	AdvancedSelector *ResourceSelector `json:"advancedSelector,omitempty"`
```

Comment:
From this name, I am suspecting there is a `BasicSelector`. Is `AdvancedSelector` name good here?

Reply:
Or rest of the selectors are `BasicSelector`?

Reply:
If I use multiple selectors will that work?

Reply:
Rest of the selectors are basic ones. The basic ones are: Name, Label & Annotation based selector. One can make use of basic ones along with this advanced selector. 

Note that advanced selector can be used to match by name, label & annotation selections as well.

Reply:
> If I use multiple selectors will that work?

yes

Reply:
> Or rest of the selectors are `BasicSelector`?

Yes